app/components/primary_navigation_component.html.erb

```diff
@@ -2,7 +2,7 @@
   <div class="govuk-width-container">
     <ul class="app-primary-navigation__list">
       <% items.each do |item| %>
-        <% if item.fetch(:current, false) || current_page?(item.fetch(:url)) || current_page?("#{item.fetch(:url)}/details") %>
+        <% if item.fetch(:current, false) || item.fetch(:comparable_urls).any? { |url| request.fullpath.include?(url) } %>
```

Comment:
I think this should be moved into a helper of some form so you can wrap some unit tests around this line. It would also help to clarify the purpose of this.